### ISSUE TITLE
the generated primes for bit-lengths 2 .. 15 are all identical

### ISSUE THREAD

**Issue by appersonj** [openssl/openssl#4656]
(in bash) try:
=========================== begin bash code ==============================
i=2
while [[ "35879" == $(openssl prime -generate -bits $i) ]] ; do echo $i; ((++i)); done
===========================end bash code ==============================

Is this intended? 
It tastes like a bug.
I notice that for bit-lengths >= 16, multiple calls get different primes

**Comment by richsalz** [comment]
Yes it's known behavior.  The prime command stops at the first prime it finds.  There are only so many small-bit primes. :)

Which version of openssl are you running?  If you're not running the latest master, then there is no good way to really seed the RNG, nore to save/restore it's state across invocations of the "prime" command.

Closing this.

**Comment by levitte** [comment]
It's version 1.0.2, FYI

**Comment by richsalz** [comment]
Then there is nothing that can be done.  The RNG seeding is not very good, and the number of bits requested is pretty small, so it is not surprising that you often get the same number.

**Comment by levitte** [comment]
@richsalz, I think you're going down the wrong path here...  RNG quality has nothing to do with this, [`BN_rand`](https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/crypto/bn/bn_prime.c#L374) is perfectly capable of delivering a number with the required amount of bits.  Now, if you ask for 2 bits, the possible numbers are 0, 1, 2 and 3, so there's not much to choose randomly from.  So it's a bit absurd to get `35879` when you ask for a 2-bit prime.

(I did a quick debug session, and sure enough, `BN_rand` did deliver a 3, and yet, the result was `35879`....  this happens consistently for all amounts of bits below 16)

Note that [`probable_prime` in 1.0.2](https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/crypto/bn/bn_prime.c#L367-L397) and [`probable_prime` in 1.1.0](https://github.com/openssl/openssl/blob/OpenSSL_1_1_0-stable/crypto/bn/bn_prime.c#L400-L480) are quite different...  as I recall, there was some development revolving very small primes in the latter, and it seems that was never backported.

**Comment by richsalz** [comment]
Maybe I misread.  Want to re-open this?

**Comment by mattcaswell** [comment]
Looks like this was discovered before and fixed by commit 96a4c31 but it was never backported to the 1.0.2 branch which has the same problem. PR #4670 does the backport.
